packages/relay/src/routes/vote.ts

```diff
@@ -0,0 +1,195 @@
+import { DB } from 'anondb/node'
+import { Express } from 'express'
+import { EpochKeyProof } from '@unirep/circuits'
+import { errorHandler } from '../middleware'
+import { UnirepSocialSynchronizer } from '../synchornizer'
+import { VoteAction } from '../types'
+
+export default (
+    app: Express,
+    db: DB,
+    synchronizer: UnirepSocialSynchronizer
+) => {
+    app.post(
+        '/api/vote',
+        errorHandler(async (req, res, next) => {
+            await Vote(req, res, db, synchronizer)
+        })
+    )
+}
+
+/**
+ * we should verify the action before we execute the transaction
+ *
+ * @param voteAction the action that the user is about to do
+ * @param findVote the vote data, if it's undefined then it's the first time to vote
+ * @returns bool whether the vote action is valid or not
+ */
+function verifyVoteAction(voteAction: VoteAction, findVote: any): boolean {
+    let result = false
+    switch (voteAction) {
+        case VoteAction.UPVOTE, VoteAction.DOWNVOTE:
+            // this epk hasn't voted
+            result = !findVote
```

Comment:
I don't understand the `!findVote` part. Is `findVote` a bool?

Reply:
I am not familiar with js, but as far as I know, this is the unique way to check if object exists or not in js.